### PR TITLE
HTTPClient Port

### PR DESCRIPTION
**Pull request by copercini**
Changed SHA1 fingerprint by Root CA verification
Changed log system
Some WiFIClientSecure fixes

### CODE REVIEW

File: libraries/HTTPClient/src/HTTPClient.h [hunk 1]
```diff
@@ -0,0 +1,220 @@
...
+public:
+    HTTPClient();
+    ~HTTPClient();
+
+    bool begin(String url);
+    bool begin(String url, const char* CAcert);
+    bool begin(String host, uint16_t port, String uri = "/");
+    bool begin(String host, uint16_t port, String uri, const char* CAcert);
+    // deprecated, use the overload above instead
+    bool begin(String host, uint16_t port, String uri, bool https, const char* CAcert)  __attribute__ ((deprecated));
```
Comment (igrr) [inline libraries/HTTPClient/src/HTTPClient.h]: What's the logic for adding a deprecated function, when adding a library for the first time?

**Comment by copercini** [comment]
I tried to keep it as closer possible to ESP8266 library, it stays so close that even the deprecated functions remain deprecated =P
Removed...

**Comment by skateone** [comment]
when is tihs going to be available in the master branch?

**Comment by proppy** [comment]
@me-no-dev @igrr did you move away from https://github.com/arduino-libraries/ArduinoHttpClient ?

**Pull request merged**